include/matx/generators/random.h

```diff
@@ -186,233 +206,62 @@ template <typename T> class [[deprecated("Use random() operator instead of rando
-  randomTensorView_t(    ShapeType &&shape,
-                         curandStatePhilox4_32_10_t *states,
-                         Distribution_t dist, T alpha, T beta)
-      : alpha_(alpha), beta_(beta)
-  {
-    dist_ = dist;
-    states_ = states;
-
-    if constexpr (RANK >= 1) {
-      strides_[RANK-1] = 1;
-    }
-
-    #pragma unroll
-    for (int i = 0; i < RANK; ++i)
-    {
-      shape_[i] = shape[i];
-    }
-
-
-    #pragma unroll
-    for (int i = RANK - 2; i >= 0; i--) {
-      strides_[i] = Stride(i+1) * Size(i+1);
-    }
-  }
-
-  __MATX_INLINE__ auto Shape() const noexcept { return shape_; }
-  /**
-   * Get the random number at an index
-   *
-   * @tparam I Unused
-   * @tparam Is Index types
-   * @return Value at index
-   */
-  template <int I = 0, typename ...Is, std::enable_if_t<I == sizeof...(Is), bool> = true>
-  constexpr __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ index_t GetValC(const cuda::std::tuple<Is...>) const {
-    return 0;
-  }
-
-  template <int I = 0, typename ...Is, std::enable_if_t<I < sizeof...(Is), bool> = true>
-  __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ index_t GetValC(const cuda::std::tuple<Is...> tup) const {
-    return GetValC<I+1, Is...>(tup) + cuda::std::get<I>(tup)*Stride(I);
-  }
+  namespace detail {
 
-  /**
-   * Retrieve a value from a random view
-   *
-   * @tparam Is Index type
-   * @param indices Index values
-   */
-  template <typename... Is>
-  inline __MATX_DEVICE__ T operator()(Is... indices) const
-  {
-    T val;
-    if constexpr (RANK == 0) {
-      get_random(val, &states_[0], dist_);
-    }
-    else {
-      get_random(val, &states_[GetValC<0, Is...>(cuda::std::make_tuple(indices...))], dist_);
-    }
-
-    return alpha_ * val + beta_;
+  template< typename T >
+  struct randIntParams{
+    T min_;
+    T max_;
   };
-
-  /**
-   * Get rank of random view
-   *
-   * @return Rank of view
-   */
-  static inline constexpr __MATX_HOST__ __MATX_DEVICE__ int32_t Rank() { return RANK; }
-
-  /**
-   *  Get size of a dimension
-   *
-   * @param dim Dimension to retrieve
-   * @return Size of dimension
-   */
-  constexpr index_t inline __MATX_HOST__ __MATX_DEVICE__ Size(int dim) const
-  {
-    return shape_[dim];
-  }
-
-  /**
-   * Get stride of a dimension
-   *
-   * @param dim Dimension to retrieve
-   * @return Stride of dimension
-   */
-  index_t inline __MATX_HOST__ __MATX_DEVICE__ Stride(int dim) const
-  {
-    return strides_[dim];
-  }
-
-  template <typename ShapeType, typename Executor>
-  __MATX_INLINE__ void PreRun([[maybe_unused]] ShapeType &&shape, [[maybe_unused]] Executor &&ex) noexcept
-  {
-  }
-
-  template <typename ShapeType, typename Executor>
-  __MATX_INLINE__ void PostRun([[maybe_unused]] ShapeType &&shape, [[maybe_unused]] Executor &&ex) noexcept  
-  {
-  }  
-};
-
-  namespace detail {
+  
+  template< typename T >
+  struct randFloatParams{
+    Distribution_t dist_;
```

Comment:
Isn't this structure only used for normal distributions?

Reply:
it's used to [scale the output of the uniform path ](https://github.com/NVIDIA/MatX/blob/c0cea227db63949de4f873b4cf2d4da18c536f37/include/matx/generators/random.h#L427)for floats as well as being used for [parameters to the random generation for the normal path](https://github.com/NVIDIA/MatX/blob/c0cea227db63949de4f873b4cf2d4da18c536f37/include/matx/generators/random.h#L430).